src/components/CourseForm/CourseForm.jsx

```diff
@@ -25,6 +34,43 @@ export default function CreateCourse({ history }) {
 		id: uuidv4(),
 	});
 
+	const isFirstRun = useRef(true);
+	useEffect(() => {
+		if (isFirstRun.current) {
+			isFirstRun.current = false;
+		} else {
+			history.push('/courses');
+			isFirstRun.current = true;
+		}
+	}, [history, allCourses]);
+
+	useEffect(() => {
+		if (auth.role === 'admin' && isUpdating) {
+			let data = allCourses.filter((course) => course.id === slug);
```

Comment:
I think Array.find is more convenient here and you can delete destructuring bellow.